snc.sh

```diff
@@ -120,30 +120,30 @@ fi
 
 # Create the INSTALL_DIR for the installer and copy the install-config
 rm -fr ${INSTALL_DIR} && mkdir ${INSTALL_DIR} && cp install-config.yaml ${INSTALL_DIR}
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml compute[0].architecture ${yq_ARCH}
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml controlPlane.architecture ${yq_ARCH}
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml baseDomain ${BASE_DOMAIN}
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml metadata.name ${CRC_VM_NAME}
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml compute[0].replicas 0
-${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml pullSecret "@HIDDEN_PULL_SECRET@"
+yq_ARCH=${yq_ARCH} ${YQ} eval --inplace '.compute[0].architecture = strenv(yq_ARCH)' ${INSTALL_DIR}/install-config.yaml
```

Comment:
It's no longer possible at all to directly use variables in the `yq` command line? Something like `${YQ} eval --inplace ".compute[0].architecture = $yq_ARCH" ${INSTALL_DIR}/install-config.yaml`
Imo this deserves a mention in the commit log

Reply:
It is not unfortunately, I can mention the docs around env variable this in the commit log.

Reply:
Note that in `${YQ} write --inplace ${INSTALL_DIR}/install-config.yaml metadata.name ${CRC_VM_NAME}`, it is the shell which replaces `$CRC_VM_NAME` with its value before running `${YQ}`, so from `YQ` perspective, there is no variable involved. Are you sure this was not some escaping-related issues that you had?
```
export SSH_KEY=foo
./yq_linux_amd64 eval ".sshKey = \"$SSH_KEY\"" ./install-config.yaml
```
adds `.sshKey=foo` to install-config.yaml

Reply:
k, the doc link that you gave says
```
While environment variables can, of course, be passed in via your CLI with string interpolation, this often comes with complex quote escaping and can be tricky to write and read. 
```

It's the fact that we need to add `yq_ARCH=${yq_ARCH}`  before the yq call which bothers me (it's because the variable is not exported that we need to do that). Maybe in our case it's good enough to use shell variable interpoalation rather than `strenv`?

Reply:
@cfergeau shell variable interpolation you mean using quote escaping?

Reply:
The shell replacing `$SSH_KEY` on the command line with the value of `$SSH_KEY` is what is called 'shell interpolation' in `yq` documentation (I think :). In order to use this instead of `strenv`, some quote escaping has to be done yes.